src/app/clusters/group-key-mgmt-server/group-key-mgmt-server.cpp

```diff
@@ -0,0 +1,410 @@
+    printStatus();
+    return EMBER_ZCL_STATUS_SUCCESS;
+}
+
+static EmberAfStatus revokeKeys(uint16_t fabricIndex, uint16_t keySetIndex)
+{
+    std::set<std::tuple<uint16_t, GroupId, uint16_t>>::iterator it = sGroupStates.begin();
+    while (it != sGroupStates.end())
+    {
+        auto curr = it++;
+        if (std::get<0>(*curr) == fabricIndex && std::get<2>(*curr) == keySetIndex)
+        {
+            sGroupStates.erase(curr);
+        }
+    }
+    printStatus();
+    return EMBER_ZCL_STATUS_SUCCESS;
+}
+
+static EmberAfStatus revokeKeys(uint16_t fabricIndex)
+{
+    std::set<std::tuple<uint16_t, GroupId, uint16_t>>::iterator it = sGroupStates.begin();
+    while (it != sGroupStates.end())
+    {
+        auto curr = it++;
+        if (std::get<0>(*curr) == fabricIndex)
+        {
+            sGroupStates.erase(curr);
+        }
+    }
+    printStatus();
+    return EMBER_ZCL_STATUS_SUCCESS;
+}
+
+//------------------------------------------------------------------------------
+// Group Key Management Cluster server callbacks
+//------------------------------------------------------------------------------
+
+/** @brief Group Key Management Cluster Server Init
+ *
+ * Server Init
+ *
+ * @param endpoint    Endpoint that is being initialized
+ */
+void emberAfGroupKeyManagementClusterServerInitCallback(chip::EndpointId endpoint) {}
+
+/**
+ * @brief  Cluster AddKey Command callback (from client)
+ */
+bool emberAfGroupKeyManagementClusterSetKeyCallback(chip::EndpointId endpoint, chip::app::CommandHandler * commandObj,
+                                                    uint16_t fabricIndex, uint16_t keySetIndex, chip::ByteSpan keyRoot,
+                                                    chip::ByteSpan epochStartTime, uint8_t securityPolicy)
+{
+    EmberAfStatus result = EMBER_ZCL_STATUS_FAILURE;
+
+    if (0 == endpoint)
+    {
+        result = setKey(fabricIndex, keySetIndex, keyRoot, epochStartTime, securityPolicy);
+    }
+    else
+    {
+        result = EMBER_ZCL_STATUS_INVALID_VALUE;
+    }
+
+    EmberStatus status = emberAfSendImmediateDefaultResponse(result);
+    if (EMBER_SUCCESS != status)
+    {
+        emberAfGroupsClusterPrintln("Group Key Management: failed to send %s: 0x%x", "default_response", status);
+    }
+    return true;
+}
+
+/**
+ * @brief  Cluster RemoveKey Command callback (from client)
+ */
+bool emberAfGroupKeyManagementClusterRemoveKeyCallback(chip::EndpointId endpoint, chip::app::CommandHandler * commandObj,
+                                                       uint16_t fabricIndex, uint16_t keySetIndex)
+{
+    EmberAfStatus result = EMBER_ZCL_STATUS_FAILURE;
+
+    if (0 == endpoint)
+    {
+        result = removeKey(fabricIndex, keySetIndex);
+    }
+    else
+    {
+        result = EMBER_ZCL_STATUS_INVALID_VALUE;
+    }
+
+    EmberStatus status = emberAfSendImmediateDefaultResponse(result);
+    if (EMBER_SUCCESS != status)
```

Comment:
What could cause failure to send? The underlying code in Matter actually does not ever directly succeed or block for determination of success. This can only fail due to a session/transport error. Not sure what the value is of printing below, given that more real errors in Matter will be later (deferred) rather than known from `emberAfSendImmediateDefaultResponse` return.